src/main.cpp

```diff
@@ -6874,10 +6225,20 @@ bool SendMessages(CNode *pto)
             // If no header would connect, or if we have too many
             // blocks, or if the peer doesn't want headers, just
             // add all to the inv queue.
-            LOCK(pto->cs_inventory);
+            std::vector<uint256> blockHashesToAnnounce;
+            {
+                TRY_LOCK(pto->cs_inventory, locked);
+                if (locked)
```

Comment:
I think when we have blocks to announce we should take the lock here, deferring is only going to slow down block prop.  In addition I have testing evidence that using try locks in SendMessages is a contributing factor to the burstiness of network traffic.